feat: Create Hacker mode theme #1988

shivalipatel6 opened the pull request:
I created a theme with a black background, black block fill for all blocks, a white shadow for all blocks (still working on putting in an actual outline), dark toolbox, dark flyout, and monospace text. I am still working on putting in green text for all block text. Need to fix CSS issue first.

<!--
  - Thanks for submitting code to Blockly!  Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## The basics

<!-- TODO: Verify the following, checking each box with an 'x' between the brackets: [x] -->

- [ X] I [validated my changes](https://developers.google.com/blockly/guides/contribute/samples#making_and_verifying_a_change)

## The details
### Resolves

<!-- TODO: What Github issue does this resolve? Please include a link. -->
Fixes #1988

### Proposed Changes
This pull request creates a new blockly theme that has a black background for everything, black fill in for blocks and a white highlight. Theme is currently incomplete as it is missing green text color, but it will be fixed later once some CSS work is fixed. 
![hacker-modescreenshot1](https://github.com/google/blockly-samples/assets/56765670/56cc1e1e-6c89-4e82-aa26-ddf27c8a9eb2)

<!-- TODO: Describe what this Pull Request does.  Include screenshots if applicable. -->

### Reason for Changes

<!--TODO: Explain why these changes should be made.  Include screenshots if applicable. -->

### Test Coverage

<!-- TODO: Please create unit tests, and explain here how they cover
           your changes, or tell us how you tested it manually. If
           your changes include browser-specific behaviour, include
           information about the browser and device that you used for
           testing. -->

### Documentation

<!-- TODO: Does any documentation need to be created or updated because of this PR?
  -        If so please explain.
  --> Once the theme is made public, the documentation about themes will need to be updated to let users know the theme exists. As well documentation about how to download the theme will be needed. 

### Additional Information

<!-- Anything else we should know? -->

shivalipatel6 committed:
feat: Create Hacker mode theme #1988

I created a theme with a black bacground, black block fill for all blocks, a white shadow for all blocks (still working on putting in an actual outline),
dark toolbox, dark flyout, and monospace text. I am still working on putting in green text for all block text. Need to fix CSS issue first.

BeksOmega commented on plugins/theme-hackermode/package.json:
```diff
@@ -0,0 +1,54 @@
+{
```
Can you mark this package as [private](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#private)?

BeksOmega commented on plugins/theme-hackermode/package.json:
Sorry the create-package script doesn't work quite correctly for first party plugins, so this file needs some manual fixes. Will list below!

BeksOmega commented on plugins/theme-hackermode/package.json:
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "blockly-theme-hackermode",
```
```suggestion
  "name": "@blockly/theme-hackermode",
```

BeksOmega commented on plugins/theme-hackermode/package.json:
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "blockly-theme-hackermode",
+  "version": "0.0.0",
+  "description": "A Blockly theme.",
```
Can you update this to be more descriptive?

BeksOmega commented on plugins/theme-hackermode/README.md:
```diff
@@ -0,0 +1,36 @@
+# blockly-theme-hackermode [![Built on Blockly](https://tinyurl.com/built-on-blockly)](https://github.com/google/blockly)
```
```suggestion
# @blockly/theme-hackermode [![Built on Blockly](https://tinyurl.com/built-on-blockly)](https://github.com/google/blockly)
```

BeksOmega commented on plugins/theme-hackermode/README.md:
````diff
@@ -0,0 +1,36 @@
+
+```js
+import * as Blockly from 'blockly';
+import Theme from 'blockly-theme-hackermode';
````
```suggestion
import Theme from '@blockly/theme-hackermode';
```

BeksOmega commented on plugins/theme-hackermode/README.md:
```diff
@@ -0,0 +1,36 @@
+  - TODO: Add theme description.
+  -->
+
+A [Blockly](https://www.npmjs.com/package/blockly) theme that ...
```
Can you update this to be more descriptive?

BeksOmega commented on plugins/theme-hackermode/README.md:
````diff
@@ -0,0 +1,36 @@
+### Yarn
+
+```
+yarn add blockly-theme-hackermode
````
```suggestion
yarn add @blockly/theme-hackermode
```

BeksOmega commented on plugins/theme-hackermode/README.md:
````diff
@@ -0,0 +1,36 @@
+### npm
+
+```
+npm install blockly-theme-hackermode --save
````
```suggestion
npm install @blockly/theme-hackermode --save
```

BeksOmega commented on plugins/theme-hackermode/GETSTARTED.md:
Please delete this file.

BeksOmega commented on plugins/theme-hackermode/package.json:
```diff
@@ -0,0 +1,54 @@
+  "peerDependencies": {
+    "blockly": "^10.2.2"
+  },
+  "publishConfig": {},
```
```suggestion
  "publishConfig": {
    "access": "public",
    "registry": "https://wombat-dressing-room.appspot.com"
  },
```

BeksOmega commented on plugins/theme-hackermode/src/index.js:
```diff
@@ -0,0 +1,116 @@
+// TODO: Edit theme overview.
+/**
+ * @fileoverview Theme overview.
+ */
```
Please complete.

BeksOmega commented on plugins/theme-hackermode/src/index.js:
```diff
@@ -0,0 +1,116 @@
+ * @fileoverview Theme overview.
+ */
+
+import Blockly from 'blockly/core';
```
```suggestion
import * as Blockly from 'blockly/core';
```

BeksOmega commented on plugins/theme-hackermode/package.json:
```diff
@@ -0,0 +1,54 @@
+    "blockly-theme",
+    "theme-hackermode"
+  ],
+  "homepage": "git@github.com:shivalipatel6/blockly-samples/tree/master/plugins/theme-hackermode#readme",
```
```suggestion
  "homepage": "git@github.com:google/blockly-samples/tree/master/plugins/theme-hackermode#readme",
```

BeksOmega commented on plugins/theme-hackermode/package.json:
```diff
@@ -0,0 +1,54 @@
+  ],
+  "homepage": "git@github.com:shivalipatel6/blockly-samples/tree/master/plugins/theme-hackermode#readme",
+  "bugs": {
+    "url": "git@github.com:shivalipatel6/blockly-samples/issues"
```
```suggestion
    "url": "git@github.com:google/blockly-samples/issues"
```

BeksOmega commented on plugins/theme-hackermode/package.json:
```diff
@@ -0,0 +1,54 @@
+  },
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:shivalipatel6/blockly-samples.git",
```
```suggestion
    "url": "git@github.com:google/blockly-samples.git",
```

BeksOmega commented on plugins/theme-hackermode/src/index.js:
```diff
@@ -0,0 +1,116 @@
+};
+
+// TODO: Edit theme name.
+export default Blockly.Theme.defineTheme('THEME_NAME', {
```
Please complete.

BeksOmega commented on plugins/theme-hackermode/src/index.js:
This file looks great =)

shivalipatel6 replied:
I tackled the issues you pointed out manually. Hopefully I did everything right. Thank you so much for helping me tackle this issue and for your patience with all the problems I kept stumbling into.

BeksOmega reviewed (changes requested):
This is looking great so far! Thank you for putting the work into this and dealing with all of the crazy rebasing =)

shivalipatel6 committed:
fix: fixing formatting errors in the hackermode theme
Made theme private, updated descriptions, and updated formatting errors in readme, pagkage.json and index.

BeksOmega commented on plugins/theme-hackermode/src/index.js:
```diff
@@ -0,0 +1,118 @@
+private:true 
```
This needs to be added to the `package.json` file as a config property!

E.g.:

```json
  "main": "./dist/index.js",
  "module": "./src/index.js",
  "unpkg": "./dist/index.js",
  // This added.
  "private": true,
  "author": "",
```

BeksOmega commented on plugins/theme-hackermode/src/index.js:
```diff
@@ -0,0 +1,118 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// TODO: Edit theme overview.
```
Can you remove this TODO now that it's completed?

BeksOmega commented on plugins/theme-hackermode/src/index.js:
```diff
@@ -0,0 +1,118 @@
+  },
+};
+
+// TODO: Edit theme name.
```
Can you remove this TODO?

BeksOmega commented on plugins/theme-hackermode/README.md:
```diff
@@ -0,0 +1,35 @@
+# @blockly/theme-hackermode [![Built on Blockly](https://tinyurl.com/built-on-blockly)](https://github.com/google/blockly)
+<!--
+  - TODO: Add theme description.
+  -->
```
Can you remove this TODO now that it's completed?

BeksOmega reviewed (changes requested):
Thanks for the fixes! Just a few more nits then this should be good to go =)

shivalipatel6 committed:
feat: Create Hacker mode theme
Deleted out To-do comments and marked the package private in the package.json file.

shivalipatel6 committed:
fix: resolve build errors in theme-hackermode
Deleted line 1 in index that privated the file so that build can run successfully.

shivalipatel6 committed:
fix: formatting issues in hackermode theme.
Ran npm ci and npm run format to fix formatting issues.

shivalipatel6 committed:
Merge branch 'google:master' into feat/hackermode

BeksOmega committed:
chore: format
